oneanddone/users/tests/test_views.py

```diff
@@ -103,3 +103,18 @@ def test_get_object_non_existent_username(self):
 
         with self.assertRaises(Http404):
             self.view.get_object()
+
+class VerifyTests(TestCase):
+    def setUp(self):
+        self.view = views.Verify()
+       
+
+    def test_redirect_url_no_profile(self):
+        """
+        If the user has no profile, redirect  the url to create user profile.
+        """
+        with patch('oneanddone.users.views.redirect') as redirect:
```

Comment:
You don't need code like this at all. You are not testing an actual redirect, you are just testing the return value of the `success_url` property. You do need to inject a user into the view, as one is expected, which is why you are seeing that error about `'Verify' object has no attribute 'user'`. Here is an example of a test that will work for the first case. Maybe you can try creating tests for the other cases in `success_url`.

```python
    def test_success_url_no_profile(self):
        """
        If the user has no profile, return the url to create user profile.
        """
        user = UserFactory.create()
        self.view.user = user
        eq_(self.view.success_url, reverse_lazy('users.profile.create'))
```